### PR TITLE
Figure dimesions improvements

### PR DESCRIPTION
**Pull request by ArupPatel**
- Added figure width in increments of 10.
- Modified configure_figure_template.
- Created menu model for figure-width menu and implemented content_menu instead of drop_down_select.

Screenshot 2021-04-06 13:53:45

### CODE REVIEW

**Comment by johanneswilm** [comment]
Hey @ArupPatel - we cannot just remove those with widths 25 and 75 as there will be many current users with documents containing those sizes and we cannot just break people's existing documents. Also, I'm not sure that it would be good idea for Fidus Writer's users to have so many choices. Could you not just expose the `figureWidthMenuModel` with the original menu items so that you can override it with a plugin but keep the options in FW the same (25, 50, 75, 100%)?

**Comment by ArupPatel** [comment]
> Hey @ArupPatel - we cannot just remove those with widths 25 and 75 as there will be many current users with documents containing those sizes and we cannot just break people's existing documents. Also, I'm not sure that it would be good idea for Fidus Writer's users to have so many choices. Could you not just expose the `figureWidthMenuModel` with the original menu items so that you can override it with a plugin but keep the options in FW the same (25, 50, 75, 100%)?

hey @johanneswilm, I have exposed the figureWidthMenuModel with original menu items. Please take a look if every thing is fine.

File: fiduswriter/document/static/js/modules/editor/dialogs/figure.js [hunk 1]
```diff
@@ -324,21 +324,27 @@ export class FigureDialog {
...
-                    this.width = newValue
-                    if (this.width == "100") {
-                        alignmentSelector.setValue('center')
-                        alignmentSelector.disable()
-                        this.aligned = 'center'
-                    } else {
-                        alignmentSelector.enable()
+        const figureWidthDOM = this.dialog.dialogEl.querySelector('.figure-width')
+        figureWidthDOM.style.width = "80%"
+        figureWidthDOM.firstElementChild.innerText = `${this.width}%`
```
Comment (johanneswilm) [inline fiduswriter/document/static/js/modules/editor/dialogs/figure.js:329]: @ArupPatel Can we get the space back so it's `${this.width} %`?

File: fiduswriter/document/static/js/modules/editor/menus/figure/utils.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const figureMenuAction = (value, figureDialog) => {
+    const buttonDOM = document.querySelector(".figure-width")
+    buttonDOM.firstElementChild.innerText = `${value}%`
```
Comment (johanneswilm) [inline fiduswriter/document/static/js/modules/editor/menus/figure/utils.js:3]: same here

**Comment by johanneswilm** [comment]
@ArupPatel Ok, one little space issue, but other than hat, it is ready to be merged.

File: fiduswriter/document/static/js/modules/editor/dialogs/figure.js (hunk 1 above)
Comment (ArupPatel) [reply fiduswriter/document/static/js/modules/editor/dialogs/figure.js]: done

File: fiduswriter/document/static/js/modules/editor/menus/figure/utils.js (hunk 2 above)
Comment (ArupPatel) [reply fiduswriter/document/static/js/modules/editor/menus/figure/utils.js]: done

File: fiduswriter/document/static/js/modules/editor/menus/figure/model.js [hunk 3]
```diff
@@ -29,3 +30,48 @@ export const figureMenuModel = () => ({
...
+            type: 'action',
+            order: 2,
+            value: '50',
+            action: figureDialog => {
+                figureMenuAction("50", figureDialog)
+            },
+            selected: false
+        },
+        {
+            title: '25%',
```
[the remark this answers is not in the document]
Comment (ArupPatel) [reply fiduswriter/document/static/js/modules/editor/menus/figure/model.js]: done

**Comment by ArupPatel** [comment]
> @ArupPatel Ok, one little space issue, but other than hat, it is ready to be merged.

hey @johanneswilm, the space issue has been taken care of.

**Comment by johanneswilm** [comment]
Thanks @ArupPatel . But do you have any idea why the system says you have not signed the CLA?

**Comment by ArupPatel** [comment]
> Thanks @ArupPatel . But do you have any idea why the system says you have not signed the CLA?

I have no idea regarding that. But the CLA has my name on it, plus I tried to signed it again but the status won't change.

**Pull request merged**